src/components/sale/SaleCollectionAdminCard.tsx

```diff
@@ -341,4 +348,68 @@ const SaleCollectionAdminCard: React.VFC<{
   )
 }
 
+const DiscountCode: React.VFC<{ type: string; target: string }> = ({ type, target }) => {
+  const apolloClient = useApolloClient()
+  const [code, setCode] = useState('')
+
+  const getDiscountCode = async ({ type, target }: { type: string; target: string }) => {
+    switch (type) {
+      case 'Coupon':
+        const { data: coupon } = await apolloClient.query<
+          hasura.GET_COUPON_CODE_BY_COUPON,
+          hasura.GET_COUPON_CODE_BY_COUPONVariables
+        >({
+          query: GET_COUPON_CODE_BY_COUPON,
+          variables: {
+            id: target,
+          },
+          fetchPolicy: 'no-cache',
+        })
+        return coupon.coupon_by_pk?.coupon_code.code || ''
+      case 'Voucher':
+        const { data: voucher } = await apolloClient.query<
+          hasura.GET_VOUCHER_CODE_BY_VOUCHER,
+          hasura.GET_VOUCHER_CODE_BY_VOUCHERVariables
+        >({
+          query: GET_VOUCHER_CODE_BY_VOUCHER,
+          variables: {
+            id: target,
+          },
+          fetchPolicy: 'no-cache',
+        })
+        return voucher.voucher_by_pk?.voucher_code.code || ''
+      default:
+        return ''
```

Comment:
same as above

Reply:
discount type will not only be voucher or coupon
other type like card and downPrice maybe just return empty value

Reply:
only coupons and vouchers have a discount code, so I think it shouldn't be handled in `getDiscountCode`